### ISSUE TITLE
Warning directives fail in quote environment latex build 

### ISSUE THREAD

**Issue by AlexEshoo** [sphinx-doc/sphinx#2632]
When building the pdf from sphinx generated latex, `pdflatex` will exit with nonzero error code if a `\notice{warning}` environment (or any other directive which uses the `\py@heavybox` command in `sphinx.sty`)is wrapped in a `\begin{quote}` environment and is part of an enumerated list. This can be the case with the following rst markup:

``` rst
#. This is a Line of text (indentation 3)
    .. warning:: This line is indented by 1 space relative to the previous line (hence is a quote in sphinx)
```

producing the following latex code:

``` LaTeX
\item {}
This is a Line of text (indentation 3)

\begin{quote}

\begin{notice}{warning}{Warning:}
This line is indented by 1 space relative to the previous line (hence is a quote in sphinx)
\end{notice}

\end{quote}
```

pdflatex then exits with non-zero error code. I only noticed it because of an unintentional tabbing typo, but it seems like a bug with the `quote` environment

**Comment by jfbu** [comment]
fixed at 18eb553. Can you check ok with you ?

**Comment by AlexEshoo** [comment]
This works for me, thanks!

**Comment by jfbu** [comment]
Thanks for reporting ! I am closing now.

**Comment by jfbu** [comment]
For the record the following pure LaTeX example show the problem:

``` latex
\documentclass{article}

\usepackage{framed}

\begin{document}
\begin{quote}
  \begin{framed}
    hello
  \end{framed}
\end{quote}
\end{document}
```

and as Warning directives use `framed` package the issue arose. It would arise also in a list environment but I have not understood how to write reStructuredText which puts a warning directive as unique thing in a list item. If there is something before, there will be no issue (because it all boils done to the fact that LaTeX list or quote environments check if a paragraph was done and framed does not count as creating paragraphs).

`framed` is also used for code blocks. In a manner I have not fully elucidated, the original Verbatim code from package `fancyvrb` does something which by sheer luck makes the surrounding list or quote happy. Hence the problem did not show yet with code-blocks, which are more likely to occur in a list or quote than warning admonitions.

**Comment by jfbu** [comment]
Finally I fixed it better at e323295. Else, in an enumerated list, the framed warning could be printed above the item label.
